statsmodels/tsa/statespace/sarimax.py

```diff
@@ -0,0 +1,1941 @@
+    enforce_stationarity : boolean
+        Whether or not to transform the AR parameters to enforce stationarity
+        in the autoregressive component of the model.
+    enforce_invertibility : boolean
+        Whether or not to transform the MA parameters to enforce invertibility
+        in the moving average component of the model.
+    hamilton_representation : boolean
+        Whether or not to use the Hamilton representation of an ARMA process.
+    trend : str{'n','c','t','ct'} or iterable
+        Parameter controlling the deterministic trend polynomial :math:`A(t)`.
+        See the class parameter documentation for more information.
+    polynomial_ar : array
+        Array containing autoregressive lag polynomial coefficients,
+        ordered from lowest degree to highest. Initialized with ones, unless
+        a coefficient is constrained to be zero (in which case it is zero).
+    polynomial_ma : array
+        Array containing moving average lag polynomial coefficients,
+        ordered from lowest degree to highest. Initialized with ones, unless
+        a coefficient is constrained to be zero (in which case it is zero).
+    polynomial_seasonal_ar : array
+        Array containing seasonal autoregressive lag polynomial coefficients,
+        ordered from lowest degree to highest. Initialized with ones, unless
+        a coefficient is constrained to be zero (in which case it is zero).
+    polynomial_seasonal_ma : array
+        Array containing seasonal moving average lag polynomial coefficients,
+        ordered from lowest degree to highest. Initialized with ones, unless
+        a coefficient is constrained to be zero (in which case it is zero).
+    polynomial_trend : array
+        Array containing trend polynomial coefficients, ordered from lowest
+        degree to highest. Initialized with ones, unless a coefficient is
+        constrained to be zero (in which case it is zero).
+    k_ar : int
+        Highest autoregressive order in the model, zero-indexed.
+    k_ar_params : int
+        Number of autoregressive parameters to be estimated.
+    k_diff : int
+        Order of intergration.
+    k_ma : int
+        Highest moving average order in the model, zero-indexed.
+    k_ma_params : int
+        Number of moving average parameters to be estimated.
+    k_seasonal_ar : int
+        Highest seasonal autoregressive order in the model, zero-indexed.
+    k_seasonal_ar_params : int
+        Number of seasonal autoregressive parameters to be estimated.
+    k_seasons : int
+        Number of seasons.
+    k_seasonal_diff : int
+        Order of seasonal intergration.
+    k_seasonal_ma : int
+        Highest seasonal moving average order in the model, zero-indexed.
+    k_seasonal_ma_params : int
+        Number of seasonal moving average parameters to be estimated.
+    k_trend : int
+        Order of the trend polynomial plus one (i.e. the constant polynomial
+        would have `k_trend=1`).
```

Comment:
don't you need also the trend polynomial if it has gaps ?  
similar to the other cases where you have highest order and actual number of terms.